pages/_middleware.tsx

```diff
@@ -0,0 +1,21 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+const COOKIE_KEY = 'dynamicChunkSize';
+
+export function middleware(request: NextRequest): NextResponse {
+  const response = NextResponse.next();
+  const cookieFromRequest = request.cookies[COOKIE_KEY];
+
+  if (!cookieFromRequest) {
+    const selectedForExperiment = Math.random() <= 0.2;
```

Comment:
A/B testing is currently set to 20% -- we can change this over time once we're more confident in the dynamic chunk sizes.